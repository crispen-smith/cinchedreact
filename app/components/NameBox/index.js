/**
 *
 * NameBox
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

/* eslint-disable react/prefer-stateless-function */
class NameBox extends React.Component {
  render() {
    return (
      <div className="form-row-primary">
        <label htmlFor={this.props.inputName}>
          {this.props.inputLabel}
          <input
            type="text"
            id={this.props.inputName}
            onChange={this.props.onChange}
          />
        </label>
      </div>
    );
  }
}

NameBox.propTypes = {
  inputName: PropTypes.string.isRequired,
  inputLabel: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default NameBox;
