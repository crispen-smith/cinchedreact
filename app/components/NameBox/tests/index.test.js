import React from 'react';
import renderer from 'react-test-renderer';

import NameBox from '../index';

describe('<NameBox />', () => {
  const InputName = 'TEST_NAME';
  const InputLabel = 'TEST_LABEL';
  const onChange = () => 1;

  it('Renders a NameBox', () => {
    const tree = renderer
      .create(
        <NameBox
          inputLabel={InputLabel}
          inputName={InputName}
          onChange={onChange}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
