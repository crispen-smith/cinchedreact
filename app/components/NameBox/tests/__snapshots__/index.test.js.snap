// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameBox /> Renders a NameBox 1`] = `
<div
  className="form-row-primary"
>
  <label
    htmlFor="TEST_NAME"
  >
    TEST_LABEL
    <input
      id="TEST_NAME"
      onChange={[Function]}
      type="text"
    />
  </label>
</div>
`;
